dimod/constrained.py

```diff
@@ -611,6 +611,21 @@ def fix_variables(self,
 
         return new
 
+    def flip_variable(self, v: Variable):
+        """Flip the specified variable in the objective and constraints.
+
+        Note that this may remove a constraint as being tracked as discrete.
```

Comment:
This is a bit ugly, but I understand why its necessary after some thought and can't think of a workaround. Clear enough docstring.

Reply:
In a followup PR I will be adding a
```python
>>> cqm.add_KinN_as_discrete(sum(Binary(v) for v in range(5)) == 3)  # returns the flipped variables
{0, 1}
```
because you can turn a KinN constraint into a discrete constraint with some variable flips. That feature is actually why I made this one and #1089